src/odemis/dataio/catmaid.py

```diff
@@ -0,0 +1,351 @@
+            image_base=self.stack_info["mirrors"][0]["image_base"],
+            zoom_level=zoom_level,
+            depth=depth,
+            col=x,
+            row=y,
+            file_extension=file_extension,
+            tile_width=tile_width,
+            tile_height=tile_height,
+        )
+        _, username, password = read_config_file(self.base_url, username=True, password=True)
+        try:
+            image = response_to_array(self.session.get(tile_url, auth=(username, password)))
+        except HTTPError as e:
+            if e.response.status_code == 404:
+                logging.info("Tile not found at %s (error 404), returning blank tile", tile_url)
+                image = numpy.zeros((tile_width, tile_height))
+            elif e.response.status_code == 401:
+                raise ValueError("Authentication Failed.")
+            else:
+                logging.info("Something went wrong, returning blank tile", tile_url)
+                image = numpy.zeros((tile_width, tile_height))
+
+        tile = model.DataArray(image, self.metadata.copy())
+        orig_pixel_size = self.metadata.get(model.MD_PIXEL_SIZE, (1, 1))
+        # calculate the pixel size of the tile for the zoom level
+        tile_pixel_size = tuple(ps * 2 ** zoom_level for ps in orig_pixel_size)
+        tile.metadata[model.MD_PIXEL_SIZE] = tile_pixel_size
+
+        # calculate the center of the tile
+        position = get_tile_md_pos((x, y), self.tile_shape, tile, self)
+        tile.metadata[model.MD_POS] = [p for p in position]
+
+        return tile
+
+    def getData(self):
+        """Abstract method of DataArrayShadow"""
+        pass
+
+
+class AcquisitionDataCatmaid(AcquisitionData):
+    """
+    Implements AcquisitionData for Catmaid instances
+    """
+
+    def __init__(self, base_url, project_id, stack_id):
+        data = self._createDataArrayShadows(base_url, project_id, stack_id)
+        AcquisitionData.__init__(self, tuple(data))
+
+    @staticmethod
+    def _createDataArrayShadows(base_url, project_id, stack_id):
+        """
+         Create the DataArrayShadow from the Catmaid metadata for the Catmaid instance
+         base_url (str): url where the Catmaid instance is hosted.
+         project_id (int): project id, indicating which of the projects to use at this Catmaid instance.
+         stack_id (int): stack id, indicating which of the stacks to use at this Catmaid instance.
+         return:
+             data (DataArrayShadows): DataArrayShadows representing the image
+         """
+        dtype = numpy.int16
+        stack_info = get_stack_info(base_url, project_id, stack_id)
+        # Dimension of the image volume in pixels at its original scale level.
+        original_shape = (stack_info["dimension"]["x"], stack_info["dimension"]["y"])
+        data = [DataArrayShadowPyramidalCatmaid(stack_info, base_url, original_shape, dtype, metadata=None)]
```

Comment:
drop original_shape, metadata, as they are inside stack_info.
You probably can also drop dtype, and make it constant in `DataArrayShadowPyramidalCatmaid`. BTW, the catmaid data is either JPEG or PNG, so they are probably uint8, in which case you can keep this type, instead of pretending it's uint16.